timesketch/lib/utils.py

```diff
@@ -36,87 +36,82 @@ def random_color():
 
-        reader = csv.DictReader(fh, delimiter=delimiter.decode('string_escape'))
-        csv_header = reader.fieldnames
-        missing_fields = []
-        # Validate the CSV header
-        for field in mandatory_fields:
-            if field not in csv_header:
-                missing_fields.append(field)
-        if missing_fields:
-            raise RuntimeError(
-                u'Missing fields in CSV header: {0:s}'.format(missing_fields))
-        for row in reader:
-            if u'timestamp' not in csv_header and u'datetime' in csv_header:
-                try:
-                    parsed_datetime = parser.parse(row[u'datetime'])
-                    row[u'timestamp'] = str(
-                        int(time.mktime(parsed_datetime.timetuple())))
-                except ValueError:
-                    continue
-
-            yield row
-
-def read_and_validate_redline(path):
-    """Generator for reading a Redline CSV file.
     Args:
         path: Path to the file
         delimiter: character used as a field separator
-    """
-    # Columns that must be present in the CSV file
 
-    # check if it is the right redline format
-    mandatory_fields = [u'Alert', u'Tag', u'Timestamp', u'Field', u'Summary']
+    Returns:
+        Generator of event rows
 
+    Raises:
+        RuntimeError is CSV format is unknown
+    """
     with open(path, 'rb') as fh:
-        csv.register_dialect('myDialect',
-                             delimiter=',',
-                             quoting=csv.QUOTE_ALL,
-                             skipinitialspace=True)
-        reader = csv.DictReader(fh, delimiter=',', dialect='myDialect')
-
+        reader = csv.DictReader(fh, delimiter=delimiter.decode('string_escape'))
         csv_header = reader.fieldnames
-        missing_fields = []
-        # Validate the CSV header
-        for field in mandatory_fields:
-            if field not in csv_header:
-                missing_fields.append(field)
-        if missing_fields:
-            raise RuntimeError(
-                u'Missing fields in CSV header: {0:s}'.format(missing_fields))
-        for row in reader:
-
-            dt = parser.parse(row['Timestamp'])
-            timestamp = int(time.mktime(dt.timetuple())) * 1000
-            dt_iso_format = dt.isoformat()
-            timestamp_desc = row['Field']
-
-            summary = row['Summary']
-            alert = row['Alert']
-            tag = row['Tag']
-
-            row_to_yield = {}
-            row_to_yield["message"] = summary
-            row_to_yield["timestamp"] = timestamp
-            row_to_yield["datetime"] = dt_iso_format
-            row_to_yield["timestamp_desc"] = timestamp_desc
-            row_to_yield["alert"] = alert #extra field
-            tags = [tag]
-            row_to_yield["tag"] = tags # extra field
-
-            yield row_to_yield
+        csv_dialect = get_csv_dialect(csv_header)
+
+        if u'redline' in csv_dialect:
```

Comment:
it seems that `get_csv_dialect` returns a string, not an array of strings. I think you want to use `csv_dialect == 'redline'`. Also, this will fail if `csv_dialect` is `None`.